packages/zone.js/test/closure/zone.closure.ts

```diff
@@ -72,16 +72,57 @@ const testClosureFunction = () => {
       logs.push('getZoneWith' + keyZone!.name);
       logs.push('get' + keyZone!.get('key'));
       logs.push('root' + Zone.root.name);
-      Object.keys((Zone as any).prototype).forEach(key => {
-        logs.push(key);
+      const zonePrototypeKeys = [
+        'get',
+        'getZoneWith',
+        'fork',
+        'wrap',
+        'run',
+        'runGuarded',
+        'runTask',
+        'scheduleTask',
+        'scheduleMicroTask',
+        'scheduleMacroTask',
+        'scheduleEventTask',
+        'cancelTask',
+      ];
+      zonePrototypeKeys.forEach(key => {
+        const desc = Object.getOwnPropertyDescriptor((Zone as any).prototype, key);
+        if (desc) {
```

Comment:
Would `hasOwnProperty()` be simpler?

```suggestion
        if ((Zone as any).prototype.hasOwnProperty(key)) {
```

Reply:
Yes, you are right, thanks